src/bcf/mod.rs

```diff
@@ -178,9 +184,15 @@ impl IndexedReader {
     ///
     /// * `path` - the path to open.
     pub fn from_path<P: AsRef<Path>>(path: P) -> Result<Self> {
-        match path.as_ref().to_str() {
-            Some(p) if path.as_ref().exists() => Ok(Self::new(&ffi::CString::new(p).unwrap())?),
-            _ => Err(Error::NonUnicodePath),
+        if !path.as_ref().exists() {
+            Err(Error::FileNotFound {
+                path: path.as_ref().to_owned(),
+            })
+        } else {
+            match path.as_ref().to_str() {
+                Some(p) if path.as_ref().exists() => Ok(Self::new(&ffi::CString::new(p).unwrap())?),
+                _ => Err(Error::NonUnicodePath),
+            }
```

Comment:
see above

Reply:
What exactly do you mean here?

Reply:
Ah, sorry my bad; what I meant was that this can also be more concise using the `match` version I suggested for the IndexedReader.